### PR TITLE
Add Status to NodeInfo CRD

### PR DESCRIPTION
**Pull request by aggarwal0009**
**Reason for Change**:
Add Status to NodeInfo CRD to support device plugin implementation 

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [x] relevant PR labels added

**Notes**:

### CODE REVIEW

File: crd/multitenancy/api/v1alpha1/nodeinfo.go [hunk 1]
```diff
@@ -36,6 +38,17 @@ type NodeInfoSpec struct {
...
 
+// NodeInfoStatus defines the observed state of NodeInfo
+type NodeInfoStatus struct {
+	// +kubebuilder:validation:Optional
+	DeviceInfos []DeviceInfo `json:"deviceInfos,omitempty"`
+}
+
+type DeviceInfo struct {
+	DeviceType  DeviceType `json:"deviceType,omitempty"`
+	DeviceCount int        `json:"deviceCount"`
```
Comment (jaer-tsun) [inline crd/multitenancy/api/v1alpha1/nodeinfo.go:49]: is this expected count of device type x?

Comment (aggarwal0009) [reply crd/multitenancy/api/v1alpha1/nodeinfo.go:49]: yes. For device plugin implementation, we need count of device types.

File: crd/multitenancy/api/v1alpha1/nodeinfo.go [hunk 2]
```diff
@@ -36,6 +38,17 @@ type NodeInfoSpec struct {
 	VMUniqueID string `json:"vmUniqueID,omitempty"`
 }
 
+// NodeInfoStatus defines the observed state of NodeInfo
+type NodeInfoStatus struct {
```
Comment (thatmattlong) [inline crd/multitenancy/api/v1alpha1/nodeinfo.go:42]: I believe Status should be a subresource, check out NodeNetworkConfig for an example of the annotation

**Review by thatmattlong — CHANGES_REQUESTED** [review]

File: crd/multitenancy/api/v1alpha1/nodeinfo.go (hunk 2 above)
Comment (aggarwal0009) [reply crd/multitenancy/api/v1alpha1/nodeinfo.go:42]: Added subresource annotation.

**Review by thatmattlong — APPROVED** [review]

**Review by isaac-dasan — APPROVED** [review]

**Comment by rbtr** [comment]
if you want this to merge, you need to kick off the CI on it
/azp run Azure Container Networking PR

**Pull request merged**